### DISCUSSION TITLE
Output of `pip install` can not be render to PDF

### DISCUSSION THREAD

**Discussion by rgaiacs** [quarto-dev/quarto-cli#6587] [Q&A] [answered]
### Description

It's quite common to find a Jupyter Notebook with a cell to install Python packages using `pip`, for example

```
%pip install foo
```

Given [this minimal working example](https://github.com/quarto-dev/quarto-cli/files/12410410/pip-install.zip) that has a Jupyter Notebook with instructions to install local a Python package, run `quarto render index.ipynb --execute --to pdf` results in

```
Starting python3 kernel...Done

Executing 'index.ipynb'
  Cell 1/1: ''...Done

pandoc 
  to: latex
  output-file: index.tex
  standalone: true
  pdf-engine: xelatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf
  
metadata
  documentclass: scrartcl
  classoption:
    - DIV=11
    - numbers=noendperiod
  papersize: letter
  header-includes:
    - '\KOMAoption{captions}{tableheading}'
  block-headings: true
  title: MWE
  

Rendering PDF
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999993 (TeX Live 2022/dev/Debian) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  
updating tlmgr

updating existing packages
ERROR: 
compilation failed- error
Text line contains an invalid character.
l.200   Building wheel for foo (setup.py) ... -^^H
                                                   ^^Hdone 

see index.log for more information.
```

The problematic line is

```
  Building wheel for foo (setup.py) ... - done
```

Can Quarto remove the invalid character when producing the intermediary `.tex`?

I tried using `pip install ./foo --progress-bar off --no-color` and encounter the same error. Also, I tried using `lualatex` instead of `xelatex` and encounter the same error.

Thanks!

**Comment by mcanouil** [comment]
Why don't you use a code cell to setup dependencies which you actually don't include?

Maybe cell magic symbol `%` could be parsed and escaped for LaTeX.

Alternatively, see <https://quarto.org/docs/computations/execution-options.html#shell-commands>.

**Reply by rgaiacs** [reply]
For teaching material, include the output of `pip install` is desirable to guide learners regarding what they should expect.

Use `%pip install foo` or `!python -m pip install ./foo` produce the same output.

I'm OK for this to be close as won't fix given that `nbcovert` also fail to create the PDF for the same reason.

**Reply by mcanouil** [reply]
Indeed, since Quarto uses nbconvert and the latter has already the issue, then maybe worth to report this upstream somehow.

**Reply by rgaiacs** [reply, marked as answer]
Reported upstream as https://github.com/jupyter/nbconvert/issues/2035.
